set.mm

```diff
@@ -256002,6 +256024,118 @@ Additional definitions for (multivariate) polynomials
         |-> ( ( ( ( v evalSub w ) ` k ) ` f ) ` ( _I |` v ) ) ) } ) $.
   $}
 
+  ${
+    $d f g h i j n r $.  $d I f h i n r $.  $d R f i n r $.  $d D g i r $.
+    $d .0. i r $.  $d B f i r $.
+    mhpfval.h $e |- H = ( I mHomP R ) $.
+    mhpfval.p $e |- P = ( I mPoly R ) $.
+    mhpfval.b $e |- B = ( Base ` P ) $.
+    mhpfval.0 $e |- .0. = ( 0g ` R ) $.
+    mhpfval.d $e |- D = { h e. ( NN0 ^m I ) | ( `' h " NN ) e. Fin } $.
+    mhpfval.i $e |- ( ph -> I e. V ) $.
+    mhpfval.r $e |- ( ph -> R e. W ) $.
+    $( Value of the "homogeneous polynomial" function.  (Contributed by Steven
+       Nguyen, 25-Aug-2023.) $)
+    mhpfval $p |- ( ph -> H = ( n e. NN0 |->
```

Comment:
According to our naming convention there should be a suffix d (mhpfval -> mhpfvald, also for the following theorems), because they are in deduction form.

Reply:
I added the suffixes but note that this was based on theorems like mvrfval and psraddcl

Reply:
Actually, we add the suffix "d" when both versions are in set.mm, but for more advanced theorems, deduction form is the default, and generally only, form, so the "d" is not necessary (I would actually remove it).

Reply:
I must admit that @benjub is right, suffix "d" is not required in this case (the Methamath book is more explicit than the comment in set.mm for this convention), so it can be removed again. Sorry, @icecream17, for the inconvenience.